source/jupyter/wire/kernel.d

```diff
@@ -218,7 +292,7 @@ struct Kernel(Backend) if(isBackend!Backend) {
         }
     }
 
-    void handleExecuteRequest(Message requestMessage)  {
+    void handleExecuteRequest(Message requestMessage) @safe {
```

Comment:
You can't require this function to be `@safe`, since it calls arbitrary backend functions. The `Kernel` struct is templated, all attributes should be inferred.

Reply:
I needed @safe because I wanted to ensure the `sender` arg in `execute` adheres to `scope`.

Reply:
Then the non-templated user code has to be `@safe` (the actual backend + the main function in the example code). It's nearly always an anti-pattern to explicitly tack attributes onto templated code. I've only ever seen it be necessary due to weird compiler bugs.